core_java/twitter/src/main/java/ca/jrvs/apps/twitter/dao/TwitterDao.java

```diff
@@ -0,0 +1,136 @@
+package ca.jrvs.apps.twitter.dao;
+
+import ca.jrvs.apps.twitter.dao.helper.HttpHelper;
+import ca.jrvs.apps.twitter.dao.helper.TwitterHttpHelper;
+import ca.jrvs.apps.twitter.model.GeoLoc;
+import ca.jrvs.apps.twitter.model.Tweet;
+import ca.jrvs.apps.twitter.utils.JsonUtil;
+import java.io.IOException;
+import java.io.UnsupportedEncodingException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.URLEncoder;
+import java.nio.charset.StandardCharsets;
+import org.apache.http.HttpResponse;
+import org.apache.http.util.EntityUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Repository;
+
+@org.springframework.stereotype.Repository
+public class TwitterDao implements CrdDao<Tweet, Long> {
+
+  private static final String BASE_URL = "https://api.twitter.com/1.1/";
+  private static final String POST_URL = BASE_URL + "statuses/update.json";
+  private static final String SHOW_URL = BASE_URL + "statuses/show/";
+  private static final String DELETE_URL = BASE_URL + "statuses/destroy/";
+
+  private static final int HTTP_OK = 200;
+
+  private HttpHelper httpHelper;
+  private Logger daoLogger = LoggerFactory.getLogger(TwitterDao.class);
+
+  /**
+   * Default Constructor, gets the OAuth values for TwitterHttpHelper itself.
+   */
+  public TwitterDao() {
+    httpHelper = new TwitterHttpHelper(System.getenv("CONSUMER_TOKEN"),
+        System.getenv("CONSUMER_SECRET"),
+        System.getenv("ACCESS_TOKEN"),
+        System.getenv("ACCESS_SECRET"));
+  }
+
+  /**
+   * Alternate constructor if you want to pass in your own TwitterHttpHelper.
+   *
+   * @param helper a TwitterHttpHelper
+   */
+  @Autowired
+  public TwitterDao(HttpHelper helper) {
+    httpHelper = helper;
+  }
+
+  /**
+   * Create an entity(Tweet) to the underlying storage.
+   *
+   * @param entity entity that to be created
+   * @return created entity
+   */
+  @Override
+  public Tweet create(Tweet entity) {
+    HttpResponse response;
+    String urlParams = "";
+    String text = entity.getText();
+    GeoLoc location = entity.getLocation();
+    try {
+      urlParams = "?status=" + URLEncoder.encode(text, StandardCharsets.UTF_8.name());
+      if (location != null) {
+        float longitude = entity.getLocation().getCoordinates()[0];
+        float latitude = entity.getLocation().getCoordinates()[1];
+        urlParams += "&long="
+          + URLEncoder.encode(Float.toString(longitude), StandardCharsets.UTF_8.name())
+          + "&lat=" + URLEncoder.encode(Float.toString(latitude), StandardCharsets.UTF_8.name());
+      }
+    } catch (UnsupportedEncodingException ueex) {
+      daoLogger.error("Encoding not supported: " + StandardCharsets.UTF_8.name());
```

Comment:
do you need to throw another exception here?

Reply:
I assumed UTF-8 would always be supported, so I didn't think to throw a new exception myself. Not entirely sure what kind of exception I would throw here, besides the same UnsupportedEncodingException, or maybe RuntimeException.